scalafmt-cli/src/main/scala/org/scalafmt/cli/Cli.scala

```diff
@@ -195,50 +167,62 @@ object Cli {
       out.println("Formatting files: " + inputMethods.toList)
     }
 
-    val sbtOptions = options.copy(
-      config = options.config.copy(runner = options.config.runner.forSbt))
+    val reporter = new ScalafmtCliReporter(options)
+    val scalafmtInstance = Scalafmt
+      .create(this.getClass.getClassLoader)
+      .withReporter(reporter)
+
+    // DESNOTE(2017-05-19, pjrt): A plain, fully passed file will (try to) be
+    // formatted regardless of what it is or where it is.
+    val fqpns = inputMethods.filter { input =>
+      AbsoluteFile.fromPath(input.filename).forall { file =>
+        options.customFiles.contains(file)
+      }
+    }
+    val scalafmtInstanceIgnoreFilters =
+      if (fqpns.isEmpty) scalafmtInstance
+      else
+        Scalafmt
+          .create(this.getClass.getClassLoader)
+          .withReporter(reporter)
+          .withRespectProjectFilters(false)
+
     val termDisplay = newTermDisplay(options, inputMethods, termDisplayMessage)
-    val exitCode = new AtomicReference(ExitCode.Ok)
-    inputMethods.par.foreach { inputMethod =>
-      val inputConfig =
-        if (inputMethod.isSbt || inputMethod.isSc) sbtOptions else options
-      val code = handleFile(inputMethod, inputConfig)
-      exitCode.getAndUpdate(new UnaryOperator[ExitCode] {
-        override def apply(t: ExitCode): ExitCode =
-          ExitCode.merge(code, t)
-      })
+
+    inputMethods.foreach { inputMethod =>
```

Comment:
Unfortunately, `scalafmt-dynamic` has lower performance when it runs parallelly.
https://github.com/scalameta/scalafmt/issues/1346#issuecomment-457840603